pkgs/http/lib/src/utils.dart

```diff
@@ -68,3 +68,22 @@ Stream<T> onDone<T>(Stream<T> stream, void Function() onDone) =>
       sink.close();
       onDone();
     }));
+
+/// Caching modes for any http request in a browser.
+enum CacheMode {
+  defaultType('default'),
+  reload('reload'),
+  noStore('no-store'),
+  noCache('no-cache'),
+  forceCache('force-cache'),
+  onlyIfCached('only-if-cached');
+
+  final String cacheType;
+
+  const CacheMode(this.cacheType);
+
+  static CacheMode fromString(String cacheType) => values.firstWhere(
```

Comment:
Can use https://api.dart.dev/beta/latest/dart-core/EnumByName/byName.html

Reply:
Since we have this function so we won't need fromString at all.